service/src/main/java/tds/config/services/impl/TimeLimitConfigurationServiceImpl.java

```diff
@@ -32,14 +37,17 @@ public TimeLimitConfigurationServiceImpl(final TimeLimitConfigurationRepository
         Optional<TimeLimitConfiguration> maybeTimeLimitConfig;
 
         if (assessmentId == null) {
-            return findTimeLimitConfiguration(clientName);
+            LOG.debug("Assessment Id is null, calling method without assessment id");
+            return timeLimitConfigurationRepository.findTimeLimitConfiguration(clientName);
```

Comment:
Much like @Transactional, Spring's proxying doesn't execute annotations when executing a method from within the same class.  This prevents one @Cacheable-annotated method from calling another within this class.